OpenRA.Mods.Common/Widgets/Logic/MultiplayerLogic.cs

```diff
@@ -356,9 +356,16 @@ void RefreshServerList()
 					}
 				}
 
-				lanGames = lanGames.GroupBy(gs => gs.Address).Select(g => g.Last()).ToList();
+				var groupedLanGames = lanGames.GroupBy(gs => gs.Address).Select(g => g.Last());
+				if (groupedLanGames.Count() > 0)
```

Comment:
This will enumerate the whole collection, which is what I was trying to get you to avoid!

`AddRange` with an empty collection is fine, so you don't need a check for that.
You can propagate the null on error and no lan games with:
```csharp
if (games != null)
    games.AddRange(groupedLanGames);
else if (groupedLanGames.Any())
    games = groupedLanGames.ToList();
```